connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -717,6 +746,47 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
                 if (started)
                     updateListener.onTaskConfigUpdate(updatedTasks);
+            } else if (record.key().startsWith(RESTART_PREFIX)) {
+                String connectorName = record.key().substring(RESTART_PREFIX.length());
+                if (value.value() == null) {
+                    log.error("Ignoring restart request because it is unexpectedly null");
+                    return;
+                }
+                if (!(value.value() instanceof Map)) {
+                    log.error("Ignoring restart request because the value is not a Map but is {}", value.value().getClass());
+                    return;
+                }
+
+                Map<String, Object> valueAsMap = (Map<String, Object>) value.value();
+
+                Object failed = valueAsMap.get("only-failed");
+                if (failed == null) {
+                    log.error("Invalid data for restart request 'only-failed' field was missing");
+                    return;
+                }
+                if (!(failed instanceof Boolean)) {
+                    log.error("Invalid data for restart request 'only-failed' field should be a Boolean but is {}", failed.getClass());
+                    return;
+                }
+                boolean onlyFailed = (Boolean) failed;
+
+                Object withTasks = valueAsMap.get("include-tasks");
+                if (withTasks == null) {
+                    log.error("Invalid data for restart request 'include-tasks' field was missing");
+                    return;
+                }
+                if (!(withTasks instanceof Boolean)) {
+                    log.error("Invalid data for restart request 'include-tasks' field should be a Boolean but is {}", withTasks.getClass());
+                    return;
+                }
+                boolean includeTasks = (Boolean) withTasks;
```

Comment:
WDYT about being a bit more tolerant of missing keys? For example, we could use defaults here.

Reply:
Good idea, I fixed it. I added a default constant in KafkaConfigBackingStore class, the only thing I was unsure of if I should use the same default constant in ConnectorsResource class API or not. They seem to be 2 different world and I didn't wanted to introduce a dependency between the classes as it would expose unwanted details so right now default=false is in 2 places. Please Let me know if you have a better idea to solve this or if it's ok to have it copied in 2 places.